### ISSUE TITLE
In JsonParser.java use try-with-resource for JsonReader jsonReader = Json.createReader(new StringReader(jsonString)); in method remapGeographicCoverage(...)

### ISSUE THREAD

**Issue by rtreacy** [IQSS/dataverse#9314]

**Comment by mreekie** [comment]
Sizing:
- If this is a single file,  then this should be a 3.
- simple fix.

**Comment by qqmyers** [comment]
Using the JsonUtil.getJsonObject() or getJsonArray methods would be better than just adding a try with resources.

**Comment by pdurbin** [comment]
@bencomp are you interested in creating a PR to fix this issue? As of 4e1bc5bccd this is the line in question:

```
diff --git a/src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java b/src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java
index febb785cd9..8f3ffd2671 100644
--- a/src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java
+++ b/src/main/java/edu/harvard/iq/dataverse/util/json/JsonParser.java
@@ -682,7 +682,7 @@ public class JsonParser {
         // convert DTO to datasetField so we can back valid values.
         Gson gson = new Gson();
         String jsonString = gson.toJson(geoCoverageDTO);
-        JsonReader jsonReader = Json.createReader(new StringReader(jsonString));
+        JsonReader jsonReader = Json.createReader(new StringReader(jsonString)); // HERE
         JsonObject obj = jsonReader.readObject();
         DatasetField geoCoverageField = parseField(obj);
``` 

We have a snapshot from Sonarcloud from a couple weeks ago that also shows it: https://sonarcloud.io/code?id=IQSS_dataverse&selected=IQSS_dataverse%3Asrc%2Fmain%2Fjava%2Fedu%2Fharvard%2Fiq%2Fdataverse%2Futil%2Fjson%2FJsonParser.java&line=685

![Screen Shot 2023-09-15 at 10 44 47 AM](https://github.com/IQSS/dataverse/assets/21006/ad7d3c23-559a-4610-844d-e8b71eba2132)

![Screen Shot 2023-09-15 at 10 44 01 AM](https://github.com/IQSS/dataverse/assets/21006/307a107f-56f2-434e-97f6-e0804c94cfe5)
